Build error with newer S3 SDK version

```
Users/tyler/tiledb/TileDB/tiledb/sm/filesystem/s3.cc:145:9: error: no matching constructor for initialization of
      'Aws::Auth::AWSCredentials'
        Aws::Auth::AWSCredentials(
        ^
/usr/local/include/aws/core/auth/AWSCredentialsProvider.h:53:13: note: candidate constructor not viable: no known conversion from
      'const basic_string<[2 * ...], allocator<char>>' to 'const basic_string<[2 * ...], Aws::Allocator<char>>' for 1st argument
            AWSCredentials(const Aws::String& accessKeyId, const Aws::String& secretKey, const Aws::String& sessionToken = "") :
            ^
/usr/local/include/aws/core/auth/AWSCredentialsProvider.h:43:28: note: candidate constructor (the implicit copy constructor) not
      viable: requires 1 argument, but 2 were provided
        class AWS_CORE_API AWSCredentials
                           ^
/usr/local/include/aws/core/auth/AWSCredentialsProvider.h:43:28: note: candidate constructor (the implicit move constructor) not
      viable: requires 1 argument, but 2 were provided
/usr/local/include/aws/core/auth/AWSCredentialsProvider.h:46:13: note: candidate constructor not viable: requires 0 arguments, but 2
      were provided
            AWSCredentials()
```

SDK version info:
```bash
$ brew info aws-sdk-cpp
aws-sdk-cpp: stable 1.5.10 (bottled), HEAD
AWS SDK for C++
https://github.com/aws/aws-sdk-cpp
/usr/local/Cellar/aws-sdk-cpp/1.5.10 (15,858 files, 298.2MB) *
  Poured from bottle on 2018-10-30 at 12:42:42
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/aws-sdk-cpp.rb
```

--- a/tiledb/sm/filesystem/s3.cc
+++ b/tiledb/sm/filesystem/s3.cc
@@ -140,10 +140,11 @@ Status S3::init(const Config::S3Params& s3_config, ThreadPool* thread_pool) {
   // If the user set config variables for aws keys use them
   if (!s3_config.aws_access_key_id.empty() &&
       !s3_config.aws_secret_access_key.empty()) {
+    Aws::String access_key_id(s3_config.aws_access_key_id.c_str());
+    Aws::String secret_access_key(s3_config.aws_secret_access_key.c_str());
     client_ = Aws::MakeShared<Aws::S3::S3Client>(
         constants::s3_allocation_tag.c_str(),
-        Aws::Auth::AWSCredentials(
-            s3_config.aws_access_key_id, s3_config.aws_secret_access_key),
+        Aws::Auth::AWSCredentials(access_key_id, secret_access_key),
         config,
         Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never,
         s3_config.use_virtual_addressing_);

